RevenueCatUI/CustomerCenter/ViewModels/CustomerCenterViewModel.swift

```diff
@@ -35,15 +35,15 @@ import RevenueCat
 
     // @PublicForExternalTesting
     @Published
-    var state: CustomerCenterViewState {
+    private(set) var state: CustomerCenterViewState {
         didSet {
             if case let .error(stateError) = state {
                 self.error = stateError
             }
         }
     }
     @Published
-    var configuration: CustomerCenterConfigData?
+    private(set) var configuration: CustomerCenterConfigData?
```

Comment:
In `CustomerCenterViewModelTests.testIsLoaded()` the `configuration` field is set. That apparently still works even if it's `private`, judging from the fact that CI passed in this PR. Is that true? If so, why is that haha?

Reply:
Ah no, I guess Customer Center tests are not being run because of the absence of the compiler flag? 🤔

Reply:
Ah, you're totally right. I am now missing the mockk framework!

I [made a change](https://github.com/RevenueCat/purchases-ios/pull/4166/commits/83f12ce5e75cf2c18111103448c94c3ef1a38248) to allow this, but am unsure if it's "worth it" - the interface is already internal, so it's just for us in that sense. Thoughts?

Reply:
I think the root cause of the problem is that we're using static references to `Purchases.shared`. This means we have an implicit dependency that we cannot "invert".

I'd propose to declare a dependency on either
- something like `CustomerCenterConfigFetcher` (we do the same for `CustomerInfoFetcher`, and [I'm adding `CurrentVersionFetcher`](https://github.com/RevenueCat/purchases-ios/pull/4169/files#diff-12a1214b23dae4d7d95973b6abf435214281404b49610c9539094248d1d60a6aR30)), or 
- the existing `CustomerCenterPurchasesType` / a new `PurchasesType` 

That way we can easily provide a fake dependency in tests. What do you think?

Reply:
Yeah, I agree with @JayShortway that using `CustomerCenterPurchasesType` might be best...

Reply:
Yes, I agree, but I'd like to finish other work before doing this. Should I merge without changing the access modifiers?

Reply:
I reverted my changes so that this is as it was before.

Reply:
Yes, works for me!